nftables.go

```diff
@@ -91,17 +93,29 @@ func newInternal(family Family, table string, execer execer) (Interface, error)
 		return nil, fmt.Errorf("could not find nftables binary: %w", err)
 	}
 
-	cmd := exec.Command(nft.path, "--check", "add", "table", string(nft.family), nft.table)
-	_, err = nft.exec.Run(cmd)
+	cmd := exec.Command(nft.path, "--version")
+	out, err := nft.exec.Run(cmd)
 	if err != nil {
 		return nil, fmt.Errorf("could not run nftables command: %w", err)
 	}
+	if strings.HasPrefix(out, "nftables v0.") || strings.HasPrefix(out, "nftables v1.0.0 ") {
```

Comment:
can we build some better abstraction? 

https://github.com/kubernetes/kubernetes/blob/0ff6a00fafee08467d946ab18c7839d9704d27d5/pkg/util/iptables/iptables.go#L707-L728

so we can use the library `"k8s.io/apimachinery/pkg/util/version"`

Reply:
IMHO the current solution is better. We don't know how `nft` will present it's version in the future, but we know _exactly_ how the versions we must avoid are doing it. I have fixed at least one issue (in dns) where a too-elaborate parsing of versions caused problems (unrecognized versions).

Reply:
As an example, when I build kube-proxy as above (with go workspaces), the version becomes:
```
Kubernetes v0.0.0-master+$Format:%H$
```
Similar things are not uncommon. A local build, e.g. for fixing a bug, may get a git commit as version and be rejected even though it's better than any official version.

Reply:
So yeah, if I had put the version check into kube-proxy, I would have used `utilversion` to parse it, but it didn't seem worth it to pull in apimachinery just for this, since, as Lars notes, we know exactly what strings we're trying to avoid.

If we end up needing to do multiple version checks in the future I'd want to revisit this decision. (But hopefully we shouldn't, because of `nft --check`...)

Reply:
sgtm